adapter-manager.js

```diff
@@ -145,24 +145,97 @@ class AdapterManager extends EventEmitter {
   }
 
   /**
+   * @method getThings
+   * @returns Returns an dictionary of all of the known things.
+   *          The dictionary key corresponds to the device id.
+   */
+  getThings() {
+    var things = [];
+    for (var thingId in this.devices) {
+      things.push(this.getThing(thingId));
+    }
+    return things;
+  }
+
+  /**
+   * @method getThing
+   * @returns Returns the thing with the indicated id.
+   */
+  getThing(thingId) {
+    var device = this.getDevice(thingId);
+    if (device) {
+      return device.getThing();
+    }
+  }
+
+  /**
+   * @method getProperties
+   * @returns Retrieves all of the properties associated with the thing
+   *          identified by `thingId`.
+   */
+  getProperties(thingId) {
+    var device = this.getDevice(thingId);
+    if (device) {
+      return device.getProperties();
+    }
+  }
+
+  /**
+   * @method getProperty
+   * @returns Retrieves the property named `propertyName` from the thing
+   *          identified by `thingId`.
+   */
+  getProperty(thingId, propertyName) {
+    var device = this.getDevice(thingId);
+    if (device) {
+      return device.getProperty(propertyName);
+    }
+  }
+
+  /**
+   * @method getPropertyValue
+   * @returns Retrieves the value of the property named `propertyName` from
+   *          the thing identified by `thingId`.
+   */
```

Comment:
Ah I see. So in Web Thing terminology getProperty() actually just gets the property description from the Thing resource whereas getPropertyValue() gets the value from the Property resource.

Note that in the current [Web Thing spec](https://moziot.github.io/wot/#property-resource) getting a Property resource returns JSON which looks like the following:

```
{
  "on": true
}
```

This is subject to change of course, if we come up with something better, but just FYI this is what I'm currently implementing on top of this code.